### PR TITLE
Minor additions

### PR DESCRIPTION
**Pull request by materialschlacht**
# added functions

- planet_coords() returns a list with all planet coordinates
- moons_coords() returns a list with all planet coordinates
- id_by_moon_name() returns id of moon, similar to id_by_planet_name()

- moon_id_refresh() return a list of [moon_ids, coordinates]
- calling a destroyed moon with its former moon_id is causing errors, with this function no relogin is needed
  I've used it to catch the exception raised from the invalid moon_id and let it double check with this function

- adding points/rank to the celestial function 

- by calling "sup.solar_satellite.level" the amount of solar sats is returned
  makes empire.supply() build routines with building solar_satellites more efficient (no need to get the current
  amount from empire.ships())

- adding the outlaw status to the planet_status in the galaxy function

# added resources to various other functions

- overall less empire.resources() calls needed in various build routines, used it to calculate the missing resources
  if building is not possible without any additional empire.resources() call

### CODE REVIEW

**Comment by materialschlacht** [comment]
wasn't 100% sure how to seperate everything nicely, just ignore the Update __init__.py commits
those are the merged ones

**Comment by PiecePaperCode** [comment]
➖ Remove moon_ids_refresh
The fatal flaw that crates this issue is by the landing page not being updated after long-running times. The best solution is to listen on planet/moon ids exceptions and trigger a landing page reload. it would solve abandon planets too.
➖ Remove add_resources
I leave it to the Job of the Developer to manage Resources. I don't think adding resources to every building call is of help. Especially as there is already a build in call. is_possible. if you want to calculate missing res you need to get res separately, or after you have completed a build. or you track it yourself on memory by getting the price of the build.
➕ Added Documentation of your functions
➕ Added tests for your functions

**Comment by materialschlacht** [comment]
good points, I totally agree 
great job polishing those functions, thanks for your work ! 👍

File: README.md [hunk 1]
```diff
@@ -111,6 +113,10 @@ empire.moon_ids()                   returns list
 
 empire.moon_names()                 returns list
 
+empire.moon_coords()                returns list
+
+empire.id_by_planet_name('name')    returns int
```
Comment (materialschlacht) [inline README.md:118]: *empire.id_by_moon_name('name')

**Pull request merged**